CMSgov/design-system
# [WNMGDS-1884] Remove deprecated props and supporting code/comments

## Summary

[Jira ticket](https://jira.cms.gov/browse/WNMGDS-1884)

- Remove deprecated props from list in Jira ticket.
- Remove comments/instances where prop was used
- Remove some of our PROD typechecking errors - TS handles this so it doesn't feel necessary for us to check in the component

## How to test

CI should pass for this work, but if you need to confirm locally...
1. Pull this branch down 
2. Run `yarn lint && yarn type-check`
3. Run `yarn test:unit` to make sure tests don't break and deprecated prop warnings don't appear in terminal

## Checklist




### If this is a change to code:


### If this is a change to documentation:
